### DISCUSSION TITLE
I dont think there is an easy way to host Medusa js application

### DISCUSSION THREAD

**Discussion by AaroneGeorge** [community/community#114707] [Programming Help] [answered]
### Body

I think its almost impossible to host a medusa js application. We have tried numerous ways to host our medusa js app for over a month, there are issues like login functionality not working or some other issues... 

If there is an easy or efficient way to host medusa js applications please tell us, your help would be much appreciated since It's been an irritating task for us.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by AaroneGeorge** [comment]
Tried to deploy the app on railway, this is the error we got: 
npm WARN config production Use `--omit=dev` instead.

> medusa-starter-default@0.0.1 clean

> cross-env ./node_modules/.bin/rimraf dist

npm WARN config production Use `--omit=dev` instead.

> medusa-starter-default@0.0.1 build:server

> cross-env npm run clean && tsc -p tsconfig.server.json

npm WARN config production Use `--omit=dev` instead.

> medusa-starter-default@0.0.1 clean

> cross-env ./node_modules/.bin/rimraf dist

npm WARN config production Use `--omit=dev` instead.

> medusa-starte
npm WARN config production Use `--omit=dev` instead.

> medusa-starter-default@0.0.1 build

> cross-env npm run clean && npm run build:server && npm run build:admin

npm WARN config production Use `--omit=dev` instead.

> medusa-starter-default@0.0.1 clean

> cross-env ./node_modules/.bin/rimraf dist

npm WARN config production Use `--omit=dev` instead.

> medusa-starter-default@0.0.1 build:server

> cross-env npm run clean && tsc -p tsconfig.server.json

npm WARN config production Use `--omit=d

**Comment by saifeddineelhanoune** [comment, marked as answer]
To host a Medusa.js application ensure you deploy the Medusa backend first as both the admin and storefront require a deployed backend URL Choose a hosting provider that supports Node.js servers like Heroku DigitalOcean or Railway and ensure it offers at least 2GB of RAM for optimal performance Your backend will connect to PostgreSQL and Redis databases which most hosting providers can manage alongside your Medusa backend For the admin ensure your hosting provider supports high RAM usage and follow the specific deployment guides for your chosen platform For the storefront deployment depends on the tools you use to create it with options like Vercel for Next.js projects.

**Reply by AaroneGeorge** [reply]
Ok will follow this for sure! Do you recommend any tutorials or any resources?

**Comment by blueholding** [comment]
any tutorial plz ?

**Reply by AaroneGeorge** [reply]
Try this [Tutorial](https://youtu.be/XjMWSwoAOQc?si=5rgpuzl6iHyeDhd1) ser.
